digdag-standards/src/main/java/io/digdag/standards/command/EcsCommandExecutor.java

```diff
@@ -386,6 +388,29 @@ else if (defaultCommandTaskTTL.isPresent() && isRunningLongerThanTTL(previousSta
         return EcsCommandStatus.of(false, nextStatus);
     }
 
+    @VisibleForTesting
+    static Optional<String> getErrorMessageFromTask(String cluster, String taskArn, EcsClient client)
+    {
+        Optional<String> errorMessage;
+        try {
+            final Task task = client.getTask(cluster, taskArn);
+            if (task.getContainers().size() > 0) {
+                errorMessage = Optional.of(task.getContainers().get(0).getReason());
+            }
+            else {
+                errorMessage = Optional.of("No container information");
+            }
+        }
+        catch (TaskSetNotFoundException e) {
+            errorMessage = Optional.of(e.getErrorMessage());
+
+        }
+        catch (RuntimeException re) {
```

Comment:
Maybe this RuntimeException doesn't contain the root cause of the task failure. We need to catch RuntimeException here?

Reply:
Ah good point. I am wondering this process.
When this method is called, the task already finished and the cause of RuntimeException is
```
final Task task = client.getTask(cluster, taskArn);
```
If we don't catch the RuntimeException, task will fail because of only getting information from AWS even though task have already finished.
But I re-considered and your pointing out may be better.
I will change it.
Thanks for your pointing out.

Reply:
I removed `catch (RuntimeException..)` and fixed the related test.